### PR TITLE
Update cl-dot.lisp

### PR DESCRIPTION
**Pull request by ko56**
Edit of function dot-graph to (automatically) compute an output format based on the suffix of the output file.

### CODE REVIEW

**Comment by rpgoldman** [comment]
At the same time we do this, should we change the default type to PDF?  Postscript is pretty much dead now, in favor of PDF.

File: cl-dot.lisp [hunk 1]
```diff
@@ -185,10 +185,11 @@ from OBJECTS, using the GRAPH-OBJECT- protocol.")
 (defun dot-graph (graph outfile &key (format :ps) (directed t))
   "Renders GRAPH to OUTFILE by running the program in \*DOT-PATH* or
 *NEATO-PATH* depending on the value of the DIRECTED keyword
-argument.  The default is a directed graph.  The default
-FORMAT is Postscript."
+argument.  The default is a directed graph. 
+The output file is assigned a suffix based on the FORMAT.
+The default FORMAT is Postscript."
   (when (null format) (setf format :ps))
```
Comment (rpgoldman) [inline cl-dot.lisp:191]: This line is unnecessary.  Unless the caller explicitly writes `(dot-graph foo "/my/out/file.xxx" :format nil)`, any failure to specify a format will get the default value.

File: cl-dot.lisp [hunk 2]
```diff
@@ -185,10 +185,11 @@ from OBJECTS, using the GRAPH-OBJECT- protocol.")
 (defun dot-graph (graph outfile &key (format :ps) (directed t))
```
Comment (rpgoldman) [inline cl-dot.lisp:185]: Suggest we modify to `(format :pdf)` -- hardly anyone uses postscript any more.

**Review by rpgoldman — CHANGES_REQUESTED** [review]
This looks great -- we just might as well fix a couple of other things at the same time.

File: cl-dot.lisp [hunk 3]
```diff
@@ -185,10 +185,11 @@ from OBJECTS, using the GRAPH-OBJECT- protocol.")
...
-argument.  The default is a directed graph.  The default
-FORMAT is Postscript."
+argument.  The default is a directed graph. 
+The output file is assigned a suffix based on the FORMAT.
+The default FORMAT is Postscript."
   (when (null format) (setf format :ps))
-
+  (setf outfile (merge-pathnames (parse-namestring outfile) (make-pathname :type (string-downcase format))))
   (let ((dot-path (if directed *dot-path* *neato-path*))
         (format (format nil "-T~(~a~)" format))
```
Comment (rpgoldman) [inline cl-dot.lisp:194]: Minor nit -- aren't these lines incorrectly indented?

Comment (rpgoldman) [reply cl-dot.lisp:194]: Also suggest replace without destructive modification of parameter variable.

**Comment by rpgoldman** [comment]
@ko56 Please review my mods.

**Review by ko56 — COMMENTED** [review]
Reviewed. I also suggested a fix for not modifying the `outfile` parameter. I hope you can see it somewhere. Github is confusing me.

**Comment by rpgoldman** [comment]
I think we are good to go. The only thing that bothers me a little is that I don't have a test for this, and it's not a function that I ever use myself: I always just print the DOT graph and then invoke `dot` from the command line to create a graphical output file.

@ko56 Would you mind pulling an update to your branch, and then running it to make sure it works for you as expected?  If it does, I will merge.

**Comment by ko56** [comment]
OK, it works for me.

**Pull request merged**